core/src/test/java/io/questdb/test/griffin/wal/WalPurgeJobTest.java

```diff
@@ -312,6 +316,119 @@ public void testOneSegment() throws Exception {
+        assertMemoryLeak(() -> {
+            // Creating a table creates a new WAL with a first segment.
+            final String tableName = testName.getMethodName();
+            compile("create table " + tableName + " as (" +
+                    "select x, " +
+                    " timestamp_sequence('2022-02-24', 1000000L) ts " +
+                    " from long_sequence(5)" +
+                    ") timestamp(ts) partition by DAY WAL");
+            assertWalExistence(true, tableName, 1);
+            assertWalLockExistence(true, tableName, 1);
+            assertSegmentExistence(true, tableName, 1, 0);
+            assertSegmentLockExistence(true, tableName, 0);
+            assertSegmentExistence(false, tableName, 1, 1);
+            assertSegmentLockExistence(false, tableName, 1);
+
+            // Create a second segment.
+            compile("alter table " + tableName + " add column sss string");
+            executeInsert("insert into " + tableName + " values (6, '2022-02-24T00:00:05.000000Z', 'x')");
+            assertWalExistence(true, tableName, 1);
+            assertSegmentExistence(true, tableName, 1, 0);
+            final File segment1DirPath = assertSegmentExistence(true, tableName, 1, 1);
+            assertSegmentExistence(false, tableName, 1, 2);
+            assertSegmentLockExistence(false, tableName, 2);
+
+            // We write a marker file to prevent the second segment "wal1/1" from being reaped.
+            final File pendingDirPath = new File(segment1DirPath, ".pending");
+            Assert.assertTrue(pendingDirPath.mkdirs());
+            final File pendingFilePath = new File(pendingDirPath, "task.pending");
+            Assert.assertTrue(pendingFilePath.createNewFile());
+
+            // Create a third segment.
+            compile("alter table " + tableName + " add column ttt string");
+            executeInsert("insert into " + tableName + " values (7, '2022-02-24T00:00:06.000000Z', 'x', 'y')");
+            assertWalExistence(true, tableName, 1);
+            assertSegmentExistence(true, tableName, 1, 0);
+            assertSegmentExistence(true, tableName, 1, 1);
+            assertSegmentExistence(true, tableName, 1, 2);
+            assertSegmentExistence(false, tableName, 1, 3);
+
+            drainWalQueue();
+            engine.releaseInactive();
+
+            assertWalExistence(true, tableName, 1);
+            assertWalLockEngagement(false, tableName, 1);
+            assertSegmentExistence(true, tableName, 1, 0);
+            assertSegmentExistence(true, tableName, 1, 1);
+            assertSegmentExistence(true, tableName, 1, 2);
+
+            runWalPurgeJob();
+
+            assertWalExistence(true, tableName, 1);
+            assertSegmentExistence(false, tableName, 1, 0); // Only the first segment is reaped.
+            assertSegmentExistence(true, tableName, 1, 1);
+            assertSegmentExistence(true, tableName, 1, 1);
```

Comment:
prob should be assertSegmentExistence(false, tableName, 1, 2)

Reply:
Yes. Done.